### DISCUSSION TITLE
Do not suggest NotImplemented for raise and except

### DISCUSSION THREAD

**Discussion by cdce8p** [microsoft/pylance-release#2662] [General]
## Environment data

-   Language Server version: 2020.10.3
-   OS and version: macOS
-   Python version: 3.9

## Expected/Suggested behaviour

Do not suggest `NotImplemented` for `raise` and `except`. It should always be `NotImplementedError`.
Unfortunately this is quite a common issue, especially with beginners.

## Code Snippet / Additional information

![Screen Shot 2020-11-04 at 22 40 14](https://user-images.githubusercontent.com/30130371/98170814-fbc56700-1eee-11eb-94dc-f7e4e7960d93.png)
![Screen Shot 2020-11-04 at 22 39 49](https://user-images.githubusercontent.com/30130371/98170808-f9630d00-1eee-11eb-842c-6e3b8982cd2f.png)

**Comment by cdce8p** [comment]
To expand on that idea:
-    Limit suggestions to classes and instances that inherit from `BaseException`
-    Expand Pylance check `Type[MyClass] is not a valid exception class` to cover `raise` and variables.

```py
class MyClass:
  pass

try:
  raise MyClass  # No error here
except MyClass:  # Type[MyClass] is not a valid exception class
  pass


my_var = None
raise my_var     # No error

**Comment by jakebailey** [comment]
We'd need to decide if it's worth only showing things that are exceptions after raise/except; one gotcha is that it's totally legal to write something like `some.module.MyException` or `Some.NestedException`, and filtering the list early would mean you wouldn't get any completions until you eventually find some object that contains an exception.
